File: app/src/main/java/com/stan/dagger_android/util/RxUtils.java
package com.stan.dagger_android.util;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by uu on 2017/12/29.
 */

public class RxUtils {

    public static  <T> ObservableTransformer<T,T> applySchedulers() {
        return new ObservableTransformer<T, T>() {
            @Override
            public ObservableSource<T> apply(Observable<T> upstream) {
                return upstream.observeOn(Schedulers.io()).subscribeOn(AndroidSchedulers.mainThread());
            }
        };
    }
}

File: app/src/main/java/com/stan/dagger_android/util/RxUtils.kt
package com.stan.dagger_android.util

import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by uu on 2017/12/29.
 */

object RxUtils {

    fun <T> applySchedulers(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream ->
            upstream
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribeOn(Schedulers.io())
        }
    }
}
